web/app/scripts/resources/query-builder-service.js

```diff
@@ -85,6 +122,19 @@
             return deferred.promise;
         }
 
+        function expandFilter(path, filter, memo) {
```

Comment:
This could use a comment -- it's a little hard to easily figure out what it's doing and why.